packages/app/src/cli/services/app/configuration.ts

```diff
@@ -0,0 +1,57 @@
+import {AppConfiguration} from '../../models/app/app.js'
+import {AppProxy} from '../dev/urls.js'
+import {getPathValue} from '@shopify/cli-kit/common/object'
+
+export function getAppProxyConfiguration(configuration: AppConfiguration): AppProxy | undefined {
+  if (!getPathValue(configuration, 'app_proxy')) return undefined
+  return {
+    proxyUrl: getPathValue(configuration, 'app_proxy.url')!,
+    proxySubPathPrefix: getPathValue(configuration, 'app_proxy.prefix')!,
+    proxySubPath: getPathValue(configuration, 'app_proxy.subpath')!,
+  }
+}
+
+export function getHomeConfiguration(configuration: AppConfiguration) {
+  return {
+    applicationUrl: getPathValue<string>(configuration, 'application_url')!,
+    embedded: getPathValue<boolean>(configuration, 'embedded')!,
+    preferencesUrl: getPathValue<string>(configuration, 'app_preferences.url'),
+  }
+}
+
+export function getPosConfiguration(configuration: AppConfiguration) {
+  return {
+    embedded: getPathValue<boolean>(configuration, 'pos.embedded'),
+  }
+}
```

Comment:
I think it'd be cleaner if you move this to the AppInterface, that way you can call it as properties instead of functions like this:

```ts
get proxyConfiguration(): AppProxy | undefined {
    if (!getPathValue(this.configuration, 'app_proxy')) return undefined
    return {
      proxyUrl: getPathValue(this.configuration, 'app_proxy.url')!,
      proxySubPathPrefix: getPathValue(this.configuration, 'app_proxy.prefix')!,
      proxySubPath: getPathValue(this.configuration, 'app_proxy.subpath')!,
    }
  }

// and then:

const proxy = app.proxyConfiguration
```